src/SFA.DAS.EAS.Portal/Database/Models/Account.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using SFA.DAS.EAS.Portal.Client.Models;
+using SFA.DAS.EAS.Portal.Types;
+
+namespace SFA.DAS.EAS.Portal.Database.Models
+{
+    public class Account : Document, IAccountDto
+    {
+        [JsonProperty("accountId")]
+        public long AccountId { get; private set; }
+
+        [JsonProperty("accountLegalEntities")]
+        public IEnumerable<IAccountLegalEntityDto> AccountLegalEntities => _accountLegalEntities;
+
+        [JsonProperty("organisations")]
+        public ICollection<Organisation> Organisations { get; set; }
+
+        [JsonProperty("outboxData")]
+        public IEnumerable<OutboxMessage> OutboxData => _outboxData;
+        
+        [JsonProperty("created")]
+        public DateTime Created { get; private set; }
+
+        [JsonProperty("updated")]
+        public DateTime? Updated { get; private set; }
+
+        [JsonProperty("deleted")]
+        public DateTime? Deleted { get; private set; }
+
+        [JsonIgnore]
+        private readonly List<AccountLegalEntity> _accountLegalEntities = new List<AccountLegalEntity>();
+                
+        [JsonIgnore]
+        private List<OutboxMessage> _outboxData = new List<OutboxMessage>();
+
+        private Account(long accountId, DateTime created, string messageId)
+            : base(1)
+        {
+            Id = Guid.NewGuid();
+            AccountId = accountId;
+            // the created date originates from event publishers, so we make sure we store it as UTC
+            Created = DateTime.SpecifyKind(created, DateTimeKind.Utc);
+            
+            AddOutboxMessage(messageId, created);
+
+            Organisations = new List<Organisation>();
+        }
+
+        // ctor for when creating account doc from reserved funding
+        // do we accept event, seems a coupling too far!
+        public Account(long accountId, long accountLegalEntityId, string legalEntityName, Guid reservationId,
+            string courseId, string courseName, DateTime startDate, DateTime endDate, DateTime created, string messageId)
+            : this(accountId, created, messageId)
+        {
+            Organisations = new List<Organisation>();
+            AddReserveFunding(accountLegalEntityId, legalEntityName, reservationId, courseId, courseName, startDate, endDate);
```

Comment:
Confusing calling AddReserveFunding in the account constructor.

Reply:
Could you expand on why you find it confusing?

This particular constructor is for creating an `Account` object from reserved funding, so to me it makes sense to call the `AddReserveFunding` method, rather than duplicating the code inside that method here.

Reply:
Because the account object isn't just for the use case of the ReserveFunding use case.

Reply:
Agreed, but classes can have more than one constructor. We can add constructors for creating the account object with whatever data we need the account to contain, at construction time.

Reply:
Out of date now due to refactoring.